src/main/kotlin/platform/mcp/fabricloom/FabricLoomDecompileSourceProvider.kt

```diff
@@ -36,7 +37,23 @@ class FabricLoomDecompileSourceProvider : AttachSourcesProvider {
         val loomData = GradleUtil.findGradleModuleData(module)?.children
             ?.find { it.key == FabricLoomData.KEY }?.data as? FabricLoomData
             ?: return emptyList()
-        return loomData.decompileTasks.map(::DecompileAction)
+
+        var env = "single"
+
+        if (loomData.splitMinecraftJar) {
+            env = if (isClientClass(psiFile)) "client" else "common"
+        }
+
+        val decompileTasks = loomData.decompileTasks[env] ?: return emptyList()
+        return decompileTasks.map(::DecompileAction)
+    }
+
+    private fun isClientClass(psiFile: PsiJavaFile): Boolean {
+        return psiFile.classes.any { clazz ->
+            val environment = clazz.getAnnotation(FabricConstants.ENVIRONMENT_ANNOTATION)
+            val value = environment?.findDeclaredAttributeValue("value")?.text
+            return value == FabricConstants.ENV_TYPE_CLIENT
```

Comment:
This doesn't work if someone puts spaces or other weird stuff in their annotations. It's also inefficient because it has to resolve the AST, which possibly involves loading and parsing the Java file if the annotation is a PSI stub. Normally you resolve the reference and check if the `PsiField` is equal to do stuff like this. However in this case, I'm fairly sure I have already implemented something in `SideOnlyUtil`?

Reply:
All of these annotations here will have been auto generated by loom when spliting the mc jar so shouldnt be weird I hope. Ill take a look at SideOnlyUtil to see if it has what I need.

I had hoped to be able to read an entry from the jar's manifest but struggled to get that to work.